### PR TITLE
Added type validation + tests. This resolves issue #80.

### PR DESCRIPTION
**Pull request by brownstein**
Pretty basic types validator passthrough.

### CODE REVIEW

File: validate.js [hunk 1]
```diff
@@ -1114,6 +1114,46 @@
...
+    type: function(value, options) {
+      // a/an selector to produce proper grammer in error messages
+      function aOrAn(nextWord) {
+        var vowels = "aeiou";
+        if (( typeof nextWord !== "string" ) || !nextWord.length) {
+          return "a";
+        }
+        return (vowels.indexOf(nextWord[0]) === -1) ? "a" : "an";
+      }
+      var typeHelperMap = {
```
Comment (jpbufe3) [inline validate.js]: The `isBoolean` and `isHash` utility functions were recently added to the library and would be easy to add here.
```hash: v.isHash,
boolean: v.isBoolean```

File: validate.js [hunk 2]
```diff
@@ -1114,6 +1114,46 @@
...
+        date:     v.isDate,
+        array:    v.isArray,
+        object:   v.isObject,
+        function: v.isFunction,
+        promise:  v.isPromise
+      };
+      if (!v.isObject(options)) {  // accept options as string or object
+        options = { type: options };
+      }
+      if (value === undefined) {  // undefined checking is already covered by 'presence'
```
Comment (jpbufe3) [inline validate.js]: You should probably also check for `null` here.
`if (value === undefined || value === null) {`

Comment (brownstein) [inline validate.js]: Addressed.

File: validate.js (hunk 1 above)
Comment (brownstein) [inline validate.js]: Addressed.

File: validate.js [hunk 3]
```diff
@@ -1114,6 +1114,48 @@
...
+        number:   v.isNumber,
+        integer:  v.isInteger,
+        string:   v.isString,
+        date:     v.isDate,
+        array:    v.isArray,
+        object:   v.isObject,
+        function: v.isFunction,
+        promise:  v.isPromise,
+        hash:     v.isHash,
+        bookean:  v.isBoolean
```
Comment (jpbufe3) [inline validate.js]: typo: `bookean` -> `boolean`

**Comment by brownstein** [comment]
At this point, I've adopted a json-schema compliant validator to meet my project's needs and am ceasing development on this branch. If 100% test coverage is a requirement for merge, I can close the PR.

File: validate.js [hunk 4]
```diff
@@ -1114,6 +1114,48 @@
...
+      };
+      if (!v.isObject(options)) {  // accept options as string or object
+        options = { type: options };
+      }
+      if (value === undefined || value === null) {  // undefined/null checking is already covered by 'presence'
```
Comment (jpbufe3) [inline validate.js]: Doesn't need to be changed again, but just realized you could probably convert this to:
`if (!v.isDefined(value)) {`

**Comment by jpbufe3** [comment]
@ansman I will open a separate PR that addresses this since @brownstein decided to close it.

**Pull request closed without merging**